### PR TITLE
[6.5.x] RHBPMS-5115 - DisabledFollowOnLockOracle10gDialect results in ORA-020…

### PR DESCRIPTION
**Pull request by mswiderski**
…14 errors

### CODE REVIEW

File: jbpm-services/jbpm-shared-services/src/main/java/org/jbpm/shared/services/impl/JpaPersistenceContext.java [hunk 1]
```diff
@@ -127,132 +134,137 @@ public int executeUpdateString(String updateString) {
...
+    
+    public <T> T merge(T entity) {
+        check();
+        return this.em.merge(entity);
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T queryStringWithParameters(Map<String, Object> params, boolean singleResult, LockModeType lockMode,
+            Class<T> clazz, Query query) {
+        ;
```
Comment (MarianMacik) [inline jbpm-services/jbpm-shared-services/src/main/java/org/jbpm/shared/services/impl/JpaPersistenceContext.java]: Unnecessary semicolon.

File: jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/jpa/ExecutorQueryServiceImpl.java [hunk 2]
```diff
@@ -228,15 +228,23 @@ public RequestInfo getRequestForProcessing(Long requestId) {
...
+	    		
+	    		if ("org.jbpm.persistence.jpa.hibernate.DisabledFollowOnLockOracle10gDialect".equals(ctx.get("hibernate.dialect"))) {
+	    		    
+	    		    Number lockedRequest = ctx.nativeQueryAndLockWithParametersInTransaction("NativePendingRequestsForProcessing",params, true, Number.class);
+	    		    if (lockedRequest != null) {
+	    		        request = ctx.find(org.jbpm.executor.entities.RequestInfo.class, lockedRequest.longValue());
+	    		    }
+	    		} else {
+	    		    params.put("firstResult", 0);
+	                params.put("maxResults", 1);
```
Comment (MarianMacik) [inline jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/jpa/ExecutorQueryServiceImpl.java]: Weird formatting, I would correct it, but it's up to you, not a big deal.

Comment (tsurdilo) [reply jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/impl/jpa/ExecutorQueryServiceImpl.java]: same small formatting on line 239 in method params

**Comment by mswiderski** [comment]
@tsurdilo @MarianMacik changes applied, plus added another 6.5.x related back port

**Pull request merged**